packages/blocks/src/api/parser/index.js

```diff
@@ -242,6 +243,13 @@ export function parseRawBlock( rawBlock ) {
 		blockType
 	);
 
+	if ( ! updatedBlock.isValid ) {
+		// Preserve the original unprocessed version of the block
+		// that we received so that we can preserve it in its
+		// existing state when we save.
+		updatedBlock.source = rawBlock;
```

Comment:
The impact here is that we're making the raw block a public API, something we never did in the client and may need deeper thoughts about impact.

cc @mtias @mcsf @WordPress/gutenberg-core

Reply:
In what other places we will need the `rawBlock`? Could we just pass the serialized raw block here?

Reply:
One thing I am not understanding is: how is this different from `updatedBlock.originalContent`? Indeed, this value is already available in `BlockListBlock` at `props.block.originalContent`.

Reply:
my first attempts were to pass the serialized raw block but we actually use various forms of the original block when performing comparisons of an "invalid block" and various ways of resolving it. further we have various ways of resolving it, some involving splitting that block into multiple blocks.

`originalContent` is fundamentally flawed in that it only looks at the `innerHTML` property of the raw block (to the neglect of inner blocks) which is the cause for a host of data loss and corruption issues and is also one of the reasons we can't convert non-nested blocks to nested blocks without creating those opportunities for data loss. technically it's the combination of our reliance on `originalContent`, `innerHTML`, and the way we call `getSaveContent(…)` to re-generate markup for invalid blocks; this PR is one step in undoing that so that we can preserve that unrecognized content and resolve issues in less risky ways.

I understand it's making this part of exposed surface of the block but we use this information, as we do with `originalContent`, in a lot of different places around the editor and I've personally concluded we have to carry it around if we want to avoid the corruption, data loss, and UX problems (such as demonstrated in the screenshot above where it looks like the list lost all its bullet points). Ultimately I would like to get rid of `originalContent` but as noted in #38922 that's a much broader change and I couldn't get it in a comprehensive yet focused PR.

Reply:
sorry I missed this comment @adamziel.

honestly I am fairly persuaded by the `source`/`__unstableSource` method, but I want to make sure we exhaust our search of other options that I may be less persuaded at initially.

in this discussion I think it's become even more clear that the block node itself is the place where this should live, and I'll rename to `__unstableSource`